Source/Core/HeadingPitchRoll.js

```diff
@@ -0,0 +1,145 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './freezeObject',
+    './Math',
+    './defineProperties'
+], function (
+        defaultValue,
+        defined,
+        DeveloperError,
+        freezeObject,
+        CesiumMath,
+        defineProperties) {
+    "use strict";
+
+    /**
+     * heading pitch roll type. Heading is the rotation about the
+     * negative z axis. Pitch is the rotation about the negative y axis. Roll is the rotation about
+     * the positive x axis.
+     * @alias HeadingPitchRoll
+     * @constructor
+     *
+     * @param {Number} [heading=0.0] The heading component in radians.
+     * @param {Number} [pitch=0.0] The pitch component in radians.
+     * @param {Number} [roll=0.0] The roll component in radians.
+     */
+    var HeadingPitchRoll = function (heading, pitch, roll) {
+        this._heading = 0.0;
+        this._pitch = 0.0;
+        this._roll = 0.0;
+
+        this.heading = defaultValue(heading, 0.0);
```

Comment:
These properties will override the ones defined on the prototype below.  Probably not what you want.

But really, the usual approach in Cesium is to allow any value (even invalid ones) to be set for primitive types, and validate the values on use where appropriate.  For example, see the `Spherical` type.  So to keep the same pattern here, the properties defined on the prototype can be removed entirely, along with the `_heading`, etc. properties defined above.